mifospay/src/main/java/org/mifospay/standinginstruction/ui/NewSIScreen.kt

```diff
@@ -0,0 +1,410 @@
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.QrCode2
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import com.google.mlkit.vision.barcode.common.Barcode
+import com.google.mlkit.vision.codescanner.GmsBarcodeScannerOptions
+import com.google.mlkit.vision.codescanner.GmsBarcodeScanning
+import org.mifospay.R
+import org.mifospay.core.designsystem.component.MfLoadingWheel
+import org.mifospay.core.designsystem.component.MifosButton
+import org.mifospay.core.designsystem.component.MifosOutlinedTextField
+import org.mifospay.core.designsystem.component.MifosScaffold
+import org.mifospay.standinginstruction.presenter.NewSIUiState
+import org.mifospay.standinginstruction.presenter.NewSIViewModel
+import org.mifospay.theme.MifosTheme
+import java.util.Calendar
+
+@Composable
+fun NewSIScreen(viewModel: NewSIViewModel = hiltViewModel(), onBackPress: () -> Unit) {
+
+    val uiState by viewModel.newSIUiState.collectAsStateWithLifecycle()
+    var cancelClicked by rememberSaveable { mutableStateOf(false) }
+
+    NewSIScreen(
+        uiState = uiState,
+        onBackPress = onBackPress,
+        fetchClient = { viewModel.fetchClient(it) },
+        cancelClicked = cancelClicked,
+        setCancelClicked = { cancelClicked = it },
+        confirm = { clientId, amount, recurrenceInterval, validTill ->
+            viewModel.createNewSI(clientId, amount, recurrenceInterval, validTill)
+        }
+    )
+
+}
+
+@Composable
+fun NewSIScreen(
+    uiState: NewSIUiState,
+    onBackPress: () -> Unit,
+    fetchClient: (String) -> Unit,
+    cancelClicked: Boolean,
+    setCancelClicked: (Boolean) -> Unit,
+    confirm: (Long, Double, Int, String) -> Unit
+) {
+    val context = LocalContext.current
+
+    MifosScaffold(
+        topBarTitle = R.string.tile_si_activity,
+        backPress = { onBackPress.invoke() },
+        scaffoldContent = {
+            when (uiState) {
```

Comment:
This flow seems wrong because always content form is visible by default and it will always be visible no matter what user action did like 
- Initially content form will be visible 
- If user click add or create etc then a progress bar will be visible on top of form but form will not go away but in below logic it will.
- We handle UI state on user action like pass, fail, validate state and show the toast etc.